docs/ref/forms/widgets.txt

```diff
@@ -410,57 +410,55 @@ foundation for custom widgets.
     :meth:`~Widget.value_from_datadict`::
 
         from datetime import date
-        from django.forms import widgets
+        from django import forms
 
-        class DateSelectorWidget(widgets.MultiWidget):
+        class DateSelectorWidget(forms.MultiWidget):
             def __init__(self, attrs=None):
-                # create choices for days, months, years
-                # example below, the rest snipped for brevity.
-                years = [(year, year) for year in (2011, 2012, 2013)]
-                _widgets = (
-                    widgets.Select(attrs=attrs, choices=days),
-                    widgets.Select(attrs=attrs, choices=months),
-                    widgets.Select(attrs=attrs, choices=years),
-                )
-                super().__init__(_widgets, attrs)
+                days = [(day, day) for day in range(1, 32)]
+                months = [(month, month) for month in range(1, 13)]
+                years = [(year, year) for year in [2018, 2019, 2020]]
+                widgets = [
+                    forms.Select(attrs=attrs, choices=days),
+                    forms.Select(attrs=attrs, choices=months),
+                    forms.Select(attrs=attrs, choices=years),
+                ]
+                super().__init__(widgets, attrs)
 
             def decompress(self, value):
-                if value:
+                if isinstance(value, date):
                     return [value.day, value.month, value.year]
+                elif isinstance(value, str):
+                    year, month, day = value.split("-")
+                    return [day, month, year]
                 return [None, None, None]
 
             def value_from_datadict(self, data, files, name):
-                datelist = [
-                    widget.value_from_datadict(data, files, name + '_%s' % i)
-                    for i, widget in enumerate(self.widgets)]
-                try:
-                    D = date(
-                        day=int(datelist[0]),
-                        month=int(datelist[1]),
-                        year=int(datelist[2]),
-                    )
-                except ValueError:
-                    return ''
-                else:
-                    return str(D)
-
-    The constructor creates several :class:`Select` widgets in a tuple. The
-    ``super`` class uses this tuple to setup the widget.
+                day, month, year = [
+                    widget.value_from_datadict(data, files, name + "_%s" % i)
+                    for i, widget in enumerate(self.widgets)
+                ]
```

Comment:
`MultiWidget` does the same and it's described below, so IMO we can simplify this with `super()`:
```python
day, month, year = super().value_from_datadict(data, files, name)
```